modules/images/image-loading-optimization/storage/rest-api.php

```diff
@@ -113,14 +113,15 @@ function ilo_register_endpoint() {
 function ilo_handle_rest_request( WP_REST_Request $request ) {
 	$post = ilo_get_url_metrics_post( $request->get_param( 'slug' ) );
 
-	$needed_minimum_viewport_widths = ilo_get_needed_minimum_viewport_widths(
+	$grouped_url_metrics = new ILO_Grouped_URL_Metrics(
 		$post ? ilo_parse_stored_url_metrics( $post ) : array(),
-		microtime( true ),
 		ilo_get_breakpoint_max_widths(),
 		ilo_get_url_metrics_breakpoint_sample_size(),
 		ilo_get_url_metric_freshness_ttl()
 	);
 
+	$needed_minimum_viewport_widths = $grouped_url_metrics->get_needed_minimum_viewport_widths( microtime( true ) );
```

Comment:
Even here I'm not sure this method is the right solution. Maybe the `foreach` loop below could be avoided if instead we just had a method like `is_group_complete( $minimum_viewport_width )`. Then that could be used here instead, as far as I can tell.